pkg/storaged/btrfs/subvolume.jsx

```diff
@@ -180,6 +181,149 @@ function subvolume_create(volume, subvol, parent_dir) {
+        } catch (err) {
+            console.warn("localstorage btrfs snapshot locations data malformed", localstorage_snapshot_data);
+            return null;
+        }
+    };
+
+    const get_localstorage_snapshot_location = subvol => {
+        const snapshot_locations = get_local_storage_snapshots_locs();
+        if (snapshot_locations != null)
+            return snapshot_locations[subvol.id] || null;
+        return snapshot_locations;
+    };
+
+    const get_current_date = async () => {
+        const out = await cockpit.spawn(["date", "+%s"]);
+        const now = parseInt(out.trim()) * 1000;
+        const d = new Date(now);
+        d.setSeconds(0);
+        d.setMilliseconds(0);
+        return d;
+    };
+
+    const folder_exists = async (path) => {
+        // Check if path exist and can be created
+        try {
+            await cockpit.spawn(["test", "-d", path]);
+            return true;
+        } catch {
+            return false;
+        }
+    };
+
+    const date = await get_current_date();
+    // Convert dates to ISO-8601
+    const current_date = date.toISOString().split("T")[0];
+    const current_date_time = date.toISOString().replace(":00.000Z", "");
+    const choices = [
+        {
+            value: "current_date",
+            title: cockpit.format(_("Current date $0"), current_date),
+        },
+        {
+            value: "current_date_time",
+            title: cockpit.format(_("Current date and time $0"), current_date_time),
+        },
+        {
+            value: "custom_name",
+            title: _("Custom name"),
+            type: "radioWithInput",
+        },
+    ];
+
+    const get_snapshot_name = (vals) => {
+        let snapshot_name = "";
+        if (vals.snapshot_name.checked == "current_date") {
+            snapshot_name = current_date;
+        } else if (vals.snapshot_name.checked === "current_date_time") {
+            snapshot_name = current_date_time;
+        } else if (vals.snapshot_name.checked === "custom_name") {
+            snapshot_name = vals.snapshot_name.inputs.custom_name;
+        }
+        return snapshot_name;
+    };
+
+    dialog_open({
+        Title: _("Create snapshot"),
+        Fields: [
+            TextInput("subvolume", _("Subvolume"),
+                      {
+                          value: subvol.pathname,
+                          disabled: true,
+                      }),
+            TextInput("snapshots_location", _("Snapshots location"),
+                      {
+                          value: get_localstorage_snapshot_location(subvol),
+                          placeholder: cockpit.format(_("Example, $0"), "/.snapshots"),
+                          explanation: (<>
+                              <p>{_("Snapshots must reside within the same btrfs volume.")}</p>
+                              <p>{_("When the snapshot location does not exist, it will be created as btrfs subvolume automatically.")}</p>
```

Comment:
Hmm, why not as a directory? Also, non existing parents should be created as well, no?